matter_server/client/models/node.py

```diff
@@ -197,14 +197,13 @@ def update(self, attributes_data: dict[str, Any]) -> None:
         for attribute_path, attribute_value in attributes_data.items():
             self.set_attribute_value(attribute_path, attribute_value)
         # extract device types from Descriptor Cluster
-        cluster = self.get_cluster(Clusters.Descriptor)
-        assert cluster is not None
-        for dev_info in cluster.deviceTypeList:  # type: ignore[unreachable]
-            device_type = DEVICE_TYPES.get(dev_info.deviceType)
-            if device_type is None:
-                LOGGER.debug("Found unknown device type %s", dev_info)
-                continue
-            self.device_types.add(device_type)
+        if cluster := self.get_cluster(Clusters.Descriptor):
+            for dev_info in cluster.deviceTypeList:  # type: ignore[unreachable]
+                device_type = DEVICE_TYPES.get(dev_info.deviceType)
+                if device_type is None:
```

Comment:
If a device type is not specified, the device will be considered unverified. Should we provide support for this type of device?

Reply:
for now yes, just to not crash. I've already emailed Switchbot about the issue.

Reply:
I'd prefer if we at least write something to the logs. Seems steep going from assert to silently ignore.

Reply:
that assert was there for the type checker. I do not see any real benefit of logging here for such an edge case.